widgetssdk/src/main/java/com/glia/widgets/view/head/controller/ServiceChatHeadController.kt

```diff
@@ -107,10 +104,22 @@ internal class ServiceChatHeadController(
     }
 
     private fun handleEngagementState(state: com.glia.widgets.engagement.State) {
-        if (state is StartedOmniCore || state is StartedCallVisualizer) {
-            newEngagementLoaded()
-        } else if (state is FinishedCallVisualizer || state is FinishedOmniCore) {
-            engagementEnded()
+        when (state) {
+            is EngagementState.StartedOmniCore, is EngagementState.StartedCallVisualizer -> {
+                newEngagementLoaded()
+            }
+
+            is EngagementState.FinishedCallVisualizer, is EngagementState.FinishedOmniCore, is EngagementState.QueueUnstaffed, is EngagementState.UnexpectedErrorHappened, is EngagementState.QueueingCanceled -> {
+                engagementEnded()
+            }
+
+            is EngagementState.Queuing, is EngagementState.PreQueuing -> {
+                queueingStarted()
+            }
```

Comment:
It might be easier for reading if each state starts from the next line, like in the `ApplicationChatHeadLayoutController`.